core/jobs/io/gcs_io.py

```diff
@@ -0,0 +1,269 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides an Apache Beam API for operating on GCS."""
+
+from __future__ import annotations
+
+from core.platform import models
+
+import apache_beam as beam
+from apache_beam.io.gcp import gcsio
+
+from typing import Dict, Optional, TypedDict, Union
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from apache_beam.io.gcp import gcsio_test
+    from mypy_imports import app_identity_services
+
+app_identity_services = models.Registry.import_app_identity_services()
+
+
+# TODO(#15613): Here we use MyPy ignore because of the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error (Class
+# cannot subclass 'PTransform' (has type 'Any')), we added an ignore here.
+class ReadFile(beam.PTransform): # type: ignore[misc]
+    """Read files form the GCS."""
+
+    def __init__(
+        self,
+        client: Optional[gcsio_test.FakeGcsClient] = None,
+        mode: str = 'r',
```

Comment:
Can this be `bool` that specify whether we want to do binary or not?

Reply:
Done